### DISCUSSION TITLE
Host/Client setup across two hosts

### DISCUSSION THREAD

**Discussion by goose-ws** [qdm12/gluetun#1463] [Q&A] [answered]
I have two hosts, `Host A` and `Host B`.

I'm wondering if it's possible that I can set up `gluetun A` on `Host A` to act as a "VPN server", and `gluetun B` on `Host B` to act as a "VPN Client". Containers (e.g. `Container B`) on `Host B` could use `gluetun B` for networking, and the `gluetun B` container would connect to the `gluetun A` container on `Host A`, where traffic to/from the VPN would ultimately take place.

So traffic would look like:

```
Container B <=> gluetun B <=> gluetun A <=> VPN
'--------Host B---------'     '----Host A-----'
```

Can gluetun be configured to work this way?

**Comment by bnhf** [comment, marked as answer]
@goose-ws 

Gluetun operates in client mode (only) for either OpenVPN or Wireguard -- so, no what you're proposing wouldn't work. However, Gluetun does have a Shadowsocks proxy server built-in. E.G., your Host B could run a Shadowsocks client container which could connect to the Gluetun client on Host A. Shadowsocks supports encryption and password protection, so it'd be OK to expose the Shadowsocks proxy port to the Internet via Host A's router.

**Reply by goose-ws** [reply]
Thanks. I'll look into solving my needs that way.

**Comment by dezza** [comment]
Hey, I found this discussion a while back I was wondering about doing something similar.. Reusing the same gluetun connection by pairing with a second wireguard server and allowing your clients to connect to your already existing "gluetun" VPN connection via a second wireguard tunnel.

My example might not be exactly what you want as I do not run 2x gluetun but gluetun+wireguard.
https://gist.github.com/dezza/ea216da3c66f9c206d5c8499a90fd932

Its a podman quadlet but it can easily be converted to docker, just find the corresponding arguments.
